Gemfile

```diff
@@ -32,6 +32,10 @@ end
 
 group :development, :test do
   gem "byebug", platforms: [:mri, :mingw, :x64_mingw]
+  gem "rspec-rails"
+  gem "capybara", ">= 2.15"
```

Comment:
gem này dúng để làm gì vậy em?

Reply:
gem capybara là một framework tự động, giúp kiểm tra các ứng dụng web bằng cách mô phỏng cách mà một người dùng thực sự tương tác với ứng dụng

Reply:
gem capybara là một framework tự động, giúp kiểm tra các ứng dụng web bằng cách mô phỏng cách mà một người dùng thực sự tương tác với ứng dụng

Reply:
- nó giúp ta check lỗi  khi ta test đúng cho controller và model rồi mà hoạt động của nó được đi đúng hay chưa
- ví dụ như khi ta cllick vào nút log in nó có hoạt động đúng để đăng nhập hay không trong khi controller đúng
- đây là đoạn code về ví du khi ta test xem hoạt động đăng nhập hoạt dộng hay không
describe "the signin process", :type => :feature do
  before :each do
    Member.make(:email => 'anhlam@gmail.com', :password => 'password')
  end

  it "signs me in" do
    visit '/sessions/new'
    within("#session") do
      fill_in 'Email', :with => 'anhlam@gmail.com'
      fill_in 'Password', :with => 'password'
    end
    click_button 'Sign in'
    expect(page).to have_content 'Success'
  end
end